### DISCUSSION TITLE
Not working with brainpool curve based encryption required for BDEW profile

### DISCUSSION THREAD

**Discussion by sbarlabanov** [phax/phase4#227] [General]
It is not possible to issue a valid signed and encrypted AS4 message using BDEW client with keys generated with `brainpoolP256r1` EC algorithm. It looks like `wss4j` and `xmlsec` do not support brainpool curves - only NIST curves (`secp256r1`, `secp384r1`...).

`brainpoolP256r1` is unfortunately the only one allowed curve for BDEW profile. 

Anybody already faced that issue? Or are we doing something wrong? :(

**Comment by phax** [comment]
The Brainpool curve is supported by the BouncyCastle library. Please check also the other discussion items labeled with BDEW - it's not my core competence - sry.

**Reply by sbarlabanov** [reply]
Thanks for points out to discussions Philip. My fault. I am closing this discussion in favour of https://github.com/phax/phase4/discussions/105. It seems to be the right place for that. So I will continue there.
